rest/data/test_results_test.go

```diff
@@ -163,10 +163,10 @@ func (s *testResultsConnectorSuite) TestFindTestResultByTestNameExists() {
 
 		tr, err := bucket.Get(s.ctx, opts.TestName)
 		s.Require().NoError(err)
-		s.NoError(tr.Close())
 
 		data, err := ioutil.ReadAll(tr)
 		s.Require().NoError(err)
+		s.NoError(tr.Close())
```

Comment:
Put this in a defer after 165, that way if line 168 fails the reader still gets closed.